cwatch.php

```diff
@@ -328,75 +320,125 @@ public function addService($package, array $vars = null, $parent_package = null,
-     * preventing the service from being suspended.
-     *
-     * @param stdClass $package A stdClass object representing the current package
-     * @param stdClass $service A stdClass object representing the current service
-     * @param stdClass $parent_package A stdClass object representing the parent
-     *  service's selected package (if the current service is an addon service)
-     * @param stdClass $parent_service A stdClass object representing the parent
-     *  service of the service being suspended (if the current service is an addon service)
-     * @return mixed null to maintain the existing meta fields or a numerically
-     *  indexed array of meta fields to be stored for this service containing:
-     *  - key The key for this meta field
-     *  - value The value for this key
-     *  - encrypted Whether or not this field should be encrypted (default 0, not encrypted)
-     * @see Module::getModule()
-     * @see Module::getModuleRow()
-     */
-    public function suspendService($package, $service, $parent_package = null, $parent_service = null) 
-    {
-        $service_fields = $this->serviceFieldsToObject($service->fields);
-        $licensekey = $service_fields->licensekey;
-        $row = $this->getModuleRow();
-        if (isset($row->meta->username)) {
-            $user = $row->meta->username;
-        }
-        if (isset($row->meta->password)) {
-            $pass = $row->meta->password;
-        }
-        if (isset($package->meta->cwatch_sandbox)){
-            $sandbox = $package->meta->cwatch_sandbox;
+        Loader::loadModels($this, ['Clients', 'Services']);
+
+        $this->validateServiceEdit($package, $vars);
+
+        if ($this->Input->errors()) {
+            return;
         }
-        $this->loadApi($user, $pass, $sandbox);
-        try {
-            if (empty($licensekey)){
-                $this->Input->setErrors(['api' => ['internal' => 'License Key not found.']]);
-            } else {
-                $response = $this->api->deactivatelicense($licensekey);
-                $json = json_decode($response->resp);
-                if ($response->success != 1){
-                    $this->Input->setErrors(['api' => ['internal' => $json[0]->message]]);
+
+        // Get cWatch API
+        $api = $this->getApi();
+        $service_fields = $this->serviceFieldsToObject($service->fields);
+        $license_keys = [];
+
+        $firstname = isset($vars['cwatch_firstname'])
+                ? $vars['cwatch_firstname']
+                : $service_fields->cwatch_firstname;
+        $lastname = isset($vars['cwatch_lastname'])
+                ? $vars['cwatch_lastname']
+                : $service_fields->cwatch_lastname;
+        $country = isset($vars['cwatch_country'])
+                ? $vars['cwatch_country']
+                : $service_fields->cwatch_country;
+
+        if (isset($vars['use_module']) && $vars['use_module'] == 'true') {
+            try {
```

Comment:
It looks like this does a bunch of stuff, which would be good for a separate method

```
if (isset($vars['use_module']) && $vars['use_module'] == 'true') {
    $fields = $this->updateService(...);
}

return $fields;
```